src/node_sqlite.cc

```diff
@@ -573,19 +642,70 @@ class BackupJob : public ThreadPoolWork {
     resolver->Reject(env()->context(), e).ToChecked();
   }
 
+  inline MaybeLocal<Object> CreateAbortError(
+      Isolate* isolate, const char* message = "The operation was aborted") {
+    Local<Value> cause;
+
+    // Extract the reason from the abort signal if available
+    if (!abort_signal_.IsEmpty()) {
+      HandleScope scope(isolate);
```

Comment:
I believe the `HandleScope` here should be moved before the `Local<Value> cause` above.